DEVELOP.md

```diff
@@ -0,0 +1,254 @@
+      - helloworldchaos    # Add this line in all pingcap.com group
+    verbs: ["*"]
+```
+HelloWorldChaos is a CRD in k8s, so we should register it, to do this, we can modify [kustomization.yaml](https://github.com/pingcap/chaos-mesh/blob/master/config/crd/kustomization.yaml), adding one line in resources section:
+```yaml
+resources:
+- bases/pingcap.com_podchaos.yaml
+- bases/pingcap.com_networkchaos.yaml
+- bases/pingcap.com_iochaos.yaml
+- bases/pingcap.com_helloworldchaos.yaml  # this is the new line
+```
+
+The last step is to implement the schema type in [api directory](https://github.com/pingcap/chaos-mesh/tree/master/api/v1alpha1). Add helloworldchaos_types.go:
+```golang
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+
+// HelloWorldChaos is the Schema for the helloworldchaos API
+type HelloWorldChaos struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// HelloWorldChaosList contains a list of HelloWorldChaos
+type HelloWorldChaosList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []HelloWorldChaos `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&HelloWorldChaos{}, &HelloWorldChaosList{})
+}
+```
+The `HelloWorldChaos` type represents the yaml content:
+```yaml
+apiVersion: pingcap.com/v1alpha1
+kind: HelloWorldChaos
+metadata:
+  name: <name-of-this-resource>
+  namespace: <ns-of-this-resource>
+```
+
+Having all these done, we can make images now:
+```
+make
+make docker-push
+```
+Note that the default `DOCKER_REGISTRY` is `localhost:5000`, which is set up by `hack/kind-cluster-build.sh`, you can overwrite it to any registry you have access permission, for development, this is enough.
+
+Before we install or upgrade chaos-mesh, we should modify [values.yaml](https://github.com/pingcap/chaos-mesh/blob/master/helm/chaos-mesh/values.yaml) of helm template, replacing the image to what we have pushed. For example, the template using `pingcap/chaos-mesh:latest` as the target image, but what we have developed is `localhost:5000/pingcap/chaos-mesh:latest`, so we should correct this:
+```yaml
+clusterScoped: true
+
+# Also see clusterScoped and controllerManager.serviceAccount
+rbac:
+  create: true
+
+controllerManager:
+  serviceAccount: chaos-controller-manager
+  ...
+  image: localhost:5000/pingcap/chaos-mesh:latest
+  ...
+chaosDaemon:
+  image: localhost:5000/pingcap/chaos-daemon:latest
+  ...
+dashboard:
+  image: localhost:5000/pingcap/chaos-dashboard:latest
+  ...
+```
+Now we can create the related custom resource type for chaos-mesh:
+```
+kubectl apply -f manifests/
+kubectl get crd podchaos.pingcap.com
+```
+Install Chaos Mesh:
+```
+helm install helm/chaos-mesh --name=chaos-mesh --namespace=chaos-testing --set chaosDaemon.runtime=containerd --set chaosDaemon.socketPath=/run/containerd/containerd.sock
```

Comment:
Can we delete the content after --set ?  Because the installation method is mentioned in helm / chaos-mesh / values.yaml. It says:
```# If you are using Kind or using containerd as CRI, you can use the
  # config below to use containerd as the runtime in chaos-daemon.
  # runtime: containerd
  # socketPath: /run/containerd/containerd.sock```

I think for developers, this should be understandable.

Reply:
For those who touch chaos-mesh the first time, it's unlikely for them to read the comment carefully. If the options are missed, they can't run network chaos correctly in kind and they will be confused.